src/Mvc/src/Microsoft.AspNetCore.Mvc.NewtonsoftJson/NewtonsoftJsonHelper.cs

```diff
@@ -17,39 +17,37 @@ namespace Microsoft.AspNetCore.Mvc.NewtonsoftJson
     /// </summary>
     internal class NewtonsoftJsonHelper : IJsonHelper
     {
-        private readonly NewtonsoftJsonOutputFormatter _jsonOutputFormatter;
-        private readonly ArrayPool<char> _charPool;
+        private readonly JsonSerializer _defaultSettingsJsonSerializer;
+        private readonly IArrayPool<char> _charPool;
 
         /// <summary>
-        /// Initializes a new instance of <see cref="NewtonsoftJsonHelper"/> that is backed by <paramref name="jsonOutputFormatter"/>.
+        /// Initializes a new instance of <see cref="NewtonsoftJsonHelper"/>.
         /// </summary>
-        /// <param name="jsonOutputFormatter">The <see cref="NewtonsoftJsonOutputFormatter"/> used to serialize JSON.</param>
+        /// <param name="options">The <see cref="MvcNewtonsoftJsonOptions"/>.</param>
         /// <param name="charPool">
         /// The <see cref="ArrayPool{Char}"/> for use with custom <see cref="JsonSerializerSettings"/> (see
         /// <see cref="Serialize(object, JsonSerializerSettings)"/>).
         /// </param>
-        public NewtonsoftJsonHelper(NewtonsoftJsonOutputFormatter jsonOutputFormatter, ArrayPool<char> charPool)
+        public NewtonsoftJsonHelper(IOptions<MvcNewtonsoftJsonOptions> options, ArrayPool<char> charPool)
         {
-            if (jsonOutputFormatter == null)
+            if (options == null)
             {
-                throw new ArgumentNullException(nameof(jsonOutputFormatter));
+                throw new ArgumentNullException(nameof(options));
             }
+
             if (charPool == null)
             {
                 throw new ArgumentNullException(nameof(charPool));
             }
 
-            _jsonOutputFormatter = jsonOutputFormatter;
-            _charPool = charPool;
+            _defaultSettingsJsonSerializer = CreateHtmlSafeSerializer(options.Value.SerializerSettings);
+            _charPool = new JsonArrayPool<char>(charPool);
         }
 
         /// <inheritdoc />
         public IHtmlContent Serialize(object value)
         {
-            var settings = ShallowCopy(_jsonOutputFormatter.PublicSerializerSettings);
-            settings.StringEscapeHandling = StringEscapeHandling.EscapeHtml;
-
-            return Serialize(value, settings);
+            return Serialize(value, _defaultSettingsJsonSerializer);
         }
 
         /// <inheritdoc />
```

Comment:
Do the inherited doc comments mention `serializerSettings.StringEscapeHandling` will be ignored? If not, add `<remarks>` there.